pennylane/ftqc/pauli_tracker.py

```diff
@@ -117,24 +121,117 @@ def pauli_prod(ops: List[Operator]) -> Operator:
+    """
+    return [(x, x ^ z)]
+
+
+def _commute_cnot(xc: int, zc: int, xt: int, zt: int):
+    r"""
+    Commute/move a Pauli represented by xz through :class:`~pennylane.CNOT`.
+
+    Args:
+        xc(int): Exponent of PauliX in the xz representation of a Pauli at the control wire.
+        zc(int): Exponent of PauliZ in the xz representation of a Pauli at the control wire.
+        xt(int): Exponent of PauliX in the xz representation of a Pauli at the target wire.
+        zt(int): Exponent of PauliZ in the xz representation of a Pauli at the target wire.
+
+    Return:
+        A list of xz tuples representing new Paulis operation that the :class:`~pennylane.cnot` commutes to.
+    """
+    return [(xc, zc ^ zt), (xc ^ xt, zt)]
+
+
+def commute_clifford_op(clifford_op: Operator, xz: List[Tuple[int, int]]) -> List[Tuple[int, int]]:
+    r"""Gets the list of xz-encoded bits representing the list of input Pauli ops after being commuted through the given Clifford op.
+    Mathematically, this function applies the following equation: :math:`new\_xz \cdot clifford\_op = clifford\_op \cdot xz`
+    up to a global phase to move the :math:`xz` through the :math:`clifford\_op` and returns the :math:`new\_xz`. Note that :math:`xz` and
+    :math:`new\_xz` represent a list of Pauli operations.
+
+    Args:
+        clifford_op (Operator): A Clifford operator class. Supported operators are: :class:`qml.S`, :class:`qml.H`, :class:`qml.CNOT`.
+        xz (list(tuple)): A list of xz tuples which map to Pauli operators
+
+    Return:
+        A list of new xz tuples that the clifford_op commute the xz to.
+
+    **Example:**
+        The following example shows how the `commute_clifford_op` works.
+
+        .. code-block:: python3
+
+            from pennylane.ftqc.pauli_tracker import commute_clifford_op
+            from pennylane import I, CNOT
+            >>> commute_clifford_op(CNOT(wires=[0,1]), [(1, 1), (1, 0)])
+            [(1, 1), (0, 0)]
+
+        A list of Pauli operators in the xz representation is returned.
+    """
+    if len(xz) != clifford_op.num_wires:
+        raise ValueError(
+            "Please ensure that the length of xz matches the number of wires of the clifford_op."
+        )
+
+    if not all(len(element) == 2 for element in xz):
+        raise ValueError(
+            "Please ensure there are 2 elements instead of in each tuple in the xz list."
+        )
+
+    xz_flatten = tuple(itertools.chain.from_iterable(xz))
+
+    if not all(element in [0, 1] for element in xz_flatten):
+        raise ValueError("Please ensure xz are either 0 or 1.")
+
+    if isinstance(clifford_op, S):
+        _x, _z = xz[0]
+        return _commute_s(_x, _z)
+
+    if isinstance(clifford_op, H):
+        _x, _z = xz[0]
+        return _commute_h(_x, _z)
+
+    if isinstance(clifford_op, CNOT):
+        _xc, _zc = xz[0]
+        _xt, _zt = xz[1]
+        return _commute_cnot(_xc, _zc, _xt, _zt)
```

Comment:
Non-blocking comment, but since you have the flattened version from above, you could pass `*xz_flattened` into all of these instead of indexing into the original `xz` to get these private variables to pass. Seems like you're just manually flattening it a second time 😅 But perhaps the current version is nicer to read.